CMakeLists.txt

```
@@ -283,14 +283,14 @@ endif()
 
 if(${CMK_USE_ZLIB})
   find_package(ZLIB REQUIRED)
-  set(CMK_SYSLIBS "$CMK_SYSLIBS -lz")
+  set(CMK_SYSLIBS "${CMK_SYSLIBS} -lz")
```

I don't think this is correct. I think we want to (also) access the bash variable `$CMK_SYSLIBS` here. Maybe:

```suggestion
  set(CMK_SYSLIBS "${CMK_SYSLIBS} $CMK_SYSLIBS -lz")
```

(same below)

---

(just to clarify, `$CMK_SYSLIBS` will be accessed through `charmc`)

---

Can you explain a bit more why it's necessary to access this variable twice? I'm a bit confused by this issue in general, I thought the fact that the variable is quoted would be sufficient.

---

> Can you explain a bit more why it's necessary to access this variable twice? I'm a bit confused by this issue in general, I thought the fact that the variable is quoted would be sufficient.

If I understand correctly, the trick here is that CMK_SYSLIBS are two variables, one for bash (`$CMK_SYSLIBS`), and one for CMake (`${CMK_SYSLIBS}`). The generated files *need* to reference `$CMK_SYSLIBS` since that gets also set in other shell files of the build system. 

Maybe it is better to write the code as follows:

```cmake
set(CMK_SYSLIBS "${CMK_SYSLIBS} -lz") # same for the other places where CMK_SYSLIBS gets set 
[...]
# after final modification of CMK_SYSLIBS
set(CMK_SYSLIBS "$CMK_SYSLIBS ${CMK_SYSLIBS}")
```